sim/firesim-lib/src/main/scala/bridges/AutoCounterBridge.scala

```diff
@@ -0,0 +1,142 @@
+package firesim
+package bridges
+
+import chisel3._
+import chisel3.util._
+import chisel3.experimental.{DataMirror, Direction}
+import chisel3.util.experimental.BoringUtils
+import freechips.rocketchip.config.{Parameters, Field}
+import freechips.rocketchip.diplomacy.AddressSet
+import freechips.rocketchip.util._
+
+import midas.widgets._
+import testchipip.{StreamIO, StreamChannel}
+
+
+class AutoCounterBundle(val numCounters: Int) extends Bundle {
+  val counters = Input(Vec(numCounters, UInt(64.W)))
+}
+
+case class AutoCounterBridgeConstArgs(numcounters: Int, autoCounterPortsMap: scala.collection.mutable.Map[Int, String], hastracerwidget: Boolean = false)
```

Comment:
Because the argument that gets passed within the transform (https://github.com/firesim/firesim/blob/c78bc1df532ecec8f6ac0a9769f908b45df4d560/sim/firesim-lib/src/main/scala/passes/AutoCounterCoverTransform.scala#L306 ) is mutable

Reply:
Can you make this immutable and then call `.toMap` in the pass when you instantiate this case class. It only needs to be mutable in the context of the pass.